node/theotherguy.go

```diff
@@ -0,0 +1,61 @@
+package node
+
+import (
+	"github.com/filecoin-project/go-filecoin/message"
+	"github.com/filecoin-project/go-filecoin/porcelain"
+	"github.com/filecoin-project/go-filecoin/repo"
+	"github.com/filecoin-project/go-filecoin/version"
+	"github.com/filecoin-project/go-filecoin/wallet"
+	bserv "github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-hamt-ipld"
+	bstore "github.com/ipfs/go-ipfs-blockstore"
+	exchange "github.com/ipfs/go-ipfs-exchange-interface"
+)
+
+// ToSplitOrNotToSplitNode is part of an ongoing refactor to cleanup `node.Node`.
+//
+// This structure will not stay, everything in here will either:
+// - be moved back to node,
+// - be moved into an existing submodule
+// - form a new submodule
+//
+// TODO: clean this up to complete the refactor https://github.com/filecoin-project/go-filecoin/issues/3140
+type ToSplitOrNotToSplitNode struct {
+	// Review: I guess this is either Node or a subsystem for managing protocols
+	VersionTable version.ProtocolVersionTable
+
+	// Review: candidate to be moved back to Node
+	PorcelainAPI *porcelain.API
+	// Repo is the repo this node was created with
+	// it contains all persistent artifacts of the filecoin node
+	// Review: candidate to be moved back to Node
+	Repo repo.Repo
+
+	// Review: I need to better understand this guy..
+	// Incoming messages for block mining.
+	Inbox *message.Inbox
+	// Messages sent and not yet mined.
+	Outbox *message.Outbox
+
+	// Review: is this its own submodule?
+	Wallet *wallet.Wallet
+
+	// Review: is this only used for storage deals or retrieval too?
+	//         if its only storage -> move to StorageSubmodule
+	//         if its both, move to SectorBuilderSubmodule?
+	// Exchange is the interface for fetching data from other nodes.
+	Exchange exchange.Interface
```

Comment:
Both deals and retrieval. But not related to sector builder. We can either gather all the storage protocol stuff into one component (including deals, retrieval, sector builder), or separate deals and retrieval from the core storage mining.

Reply:
I guess you could start an instance that only has the SectorBuilder and not the bitswap Piece Exchange running. Maybe for running some local analysis or maybe just for running some proofs (PIP, PoST).
I think they are tightly related though, but I can separate it as a `StorageNetworkingSubmodule`.